Switch repo config to use public Azure docs repos directly

## Summary

The current `config/repos.yaml` is configured to scan private `-pr` repos (e.g., `azure-docs-pr`) but we don't have access to these private repos. This causes:

1. Every scan attempt gets a 404 on the private repo
2. Code falls back to the public repo (e.g., `azure-docs`)
3. Baseline lookup fails because scan history is keyed by the original `-pr` URL
4. Result: **full rescan of 16k+ files every time** instead of incremental scans

## Investigation (Scan 95)

```
[github_discovery_service] Using baseline type: none, reason: No suitable baseline found
[ERROR] Could not get HEAD commit for MicrosoftDocs/azure-docs-pr:main: 404
[github_discovery_service] Private repo not accessible, trying public fallback: MicrosoftDocs/azure-docs
[github_discovery_service] Using baseline type: none, reason: No suitable baseline found
```

Scan 92 completed successfully with 13k files and valid `last_commit_sha`, but scan 95 couldn't use it as a baseline due to the URL mismatch during fallback.

## Current Config

```yaml
repos:
  - owner: MicrosoftDocs
    name: azure-docs-pr           # Private - we get 404
    public_name: azure-docs       # What we actually scan
    branch: main
    articles_path: articles
```

## Proposed Fix

Update `config/repos.yaml` to use public repo names directly:

```yaml
repos:
  - owner: MicrosoftDocs
    name: azure-docs              # Use public repo directly
    branch: main
    articles_path: articles
```

This eliminates the fallback complexity and allows incremental scanning to work correctly.

## Repos to Update

- `azure-docs-pr` → `azure-docs`
- `azure-management-docs-pr` → `azure-management-docs`
- `azure-compute-docs-pr` → `azure-compute-docs`

## Alternative

If we eventually need private repo access, we should configure a GitHub token with appropriate permissions and remove the fallback logic. But using public repos directly is simpler if private access isn't required.

--- a/config/repos.yaml
+++ b/config/repos.yaml
@@ -4,19 +4,21 @@
 
 repos:
   - owner: MicrosoftDocs
-    name: azure-docs-pr
-    public_name: azure-docs  # Public repo equivalent for fallback
+    name: azure-docs
     branch: main
     articles_path: articles
 
   - owner: MicrosoftDocs
-    name: azure-management-docs-pr
-    public_name: azure-management-docs
+    name: azure-management-docs
     branch: main
     articles_path: articles
 
   - owner: MicrosoftDocs
-    name: azure-compute-docs-pr
-    public_name: azure-compute-docs
+    name: azure-compute-docs
+    branch: main
+    articles_path: articles
+
+  - owner: MicrosoftDocs
+    name: azure-aks-docs
     branch: main
     articles_path: articles

